.mage/magefile.go

```diff
@@ -0,0 +1,112 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"github.com/magefile/mage/target"
+	"go.einride.tech/mage-tools/mglog"
+	"go.einride.tech/mage-tools/mgmake"
+	"go.einride.tech/mage-tools/mgpath"
+	"go.einride.tech/mage-tools/mgtool"
+	"go.einride.tech/mage-tools/targets/mgbuf"
+
+	// mage:import
+	"go.einride.tech/mage-tools/targets/mggo"
+
+	// mage:import
+	"go.einride.tech/mage-tools/targets/mggolangcilint"
+
+	// mage:import
+	"go.einride.tech/mage-tools/targets/mggoreview"
+
+	// mage:import
+	"go.einride.tech/mage-tools/targets/mgmarkdownfmt"
+
+	// mage:import
+	"go.einride.tech/mage-tools/targets/mgconvco"
+
+	// mage:import
+	"go.einride.tech/mage-tools/targets/mggitverifynodiff"
+)
+
+func init() {
+	mgmake.GenerateMakefiles(
+		mgmake.Makefile{
+			Path:          mgpath.FromGitRoot("Makefile"),
+			DefaultTarget: All,
+		},
+		mgmake.Makefile{
+			Path:          mgpath.FromGitRoot("proto/Makefile"),
+			DefaultTarget: Proto.All,
+			Namespace:     Proto{},
+		},
+	)
+}
+
+func All() {
+	mg.Deps(
+		mg.F(mgconvco.ConvcoCheck, "origin/master..HEAD"),
+		mgmarkdownfmt.FormatMarkdown,
+		GoStringer,
+		Proto.All,
+	)
+	mg.Deps(
+		mggolangcilint.GolangciLint,
+		mggoreview.Goreview,
+		mggo.GoTest,
+	)
+	mg.SerialDeps(
+		mggo.GoModTidy,
+		mggitverifynodiff.GitVerifyNoDiff,
+	)
+}
+
+func ProtocGenGoAip() error {
```

Comment:
shouldn't this be in the proto namespace?

Reply:
Based on the old Makefile, this was in the root Makefile

Reply:
ah!